src/components/Checkbox/Checkbox.scss

```diff
@@ -5,199 +5,199 @@
   cursor: pointer;
   margin-top: 8px;
   position: relative;
-
-  // The actual styled choicefield element - radio button by default
-  &:after {
-    content: '';
-    display: inline-block;
-    border: 1px $ms-color-neutralTertiaryAlt solid;
-    width: 19px;
-    height: 19px;
-    cursor: pointer;
-    position: relative;
-    font-weight: normal;
-    left: -1px;
-    top: -1px;
-    border-radius: 50%;
-    position: absolute;
-  }
+  vertical-align: top;
+  user-select: none;
 
   &:hover {
-    &:after {
+    &::before {
+      border-color: $ms-color-neutralSecondaryAlt;
+    }
+    .ms-Label {
+      color: $ms-color-black;
+    }
+  }
+  &:focus {
+        &::before {
       border-color: $ms-color-neutralSecondaryAlt;
     }
 
+    &.is-disabled::before {
+      border-color: $ms-color-neutralTertiaryAlt;
+    }
+
+    &.is-checked::before {
+      border-color: $ms-color-themeDarkAlt;
+    }
+  }
+  &:active {
+    &::before {
+      border-color: $ms-color-neutralSecondaryAlt;
+    }
     .ms-Label {
       color: $ms-color-black;
     }
   }
-}
 
-// A selected radio button
-.ms-ChoiceField-input:checked {
-  + .ms-ChoiceField-field {
-    // Circle indicating a checked radio button
-    &:before {
-      background-color: $ms-color-neutralSecondary;
-      border-color: $ms-color-neutralSecondary;
-      color: $ms-color-neutralSecondary;
-      border-radius: 50%;
-      content: '\00a0';
-      display: inline-block;
-      position: absolute;
-      top: 4px;
-      bottom: 0;
-      left: 4px;
-      width: 11px;
-      height: 11px;
-      box-sizing: border-box;
+    //== State: A checkbox is checked
+  //
+  &.is-checked {
+    &::before {
+      border: 10px solid $ms-color-themePrimary;
+      background-color: $ms-color-themePrimary;
 
       @media screen and (-ms-high-contrast: active) {
-        border-color: $ms-color-white;
-        background-color: $ms-color-white;
+        border-color: $ms-color-contrastBlackSelected;
       }
 
       @media screen and (-ms-high-contrast: black-on-white) {
-        border-color: $ms-color-black;
-        background-color: $ms-color-black;
+        border-color: $ms-color-contrastWhiteSelected;
       }
     }
 
-    &:hover:before {
-      background-color: $ms-color-neutralDark;
-      color: $ms-color-neutralDark;
+    &::after {
+      display: block;
     }
-  }
-}
 
-// Checkbox
-.ms-ChoiceField-input[type="checkbox"] {
-  + .ms-ChoiceField-field:after {
-    border-radius: 0;
+    &:hover,
+    &:focus {
+      &::before {
+        border-color: $ms-color-themeDarkAlt;
+      }
+    }
   }
-}
 
-// A selected checkbox
-.ms-ChoiceField-input[type="checkbox"]:checked {
-  + .ms-ChoiceField-field {
-    &:before {
-      @include ms-Icon;
-      content: '\e041';
-      background-color: transparent;
-      border-radius: 0;
-      font-size: $ms-font-size-s-plus;
-      top: 3px;
-      left: 3px;
+  //== State: A disabled checkbox
+  //
+  &.is-disabled {
```

Comment:
Targeting :disabled attribute will make sense on .ms-Checkbox-input, but then we will require those sibling selectors to specify styles for .ms-Checkbox-field